[CI] CancelTasksRelocationIT testCancelByOriginalTaskIdAfterRelocation failing

**Build Scans:**
- [elasticsearch-intake #40108 / part1](https://gradle-enterprise.elastic.co/s/cmnl7g5cwdrnw)
- [elasticsearch-pull-request #148321 / part-1](https://gradle-enterprise.elastic.co/s/udpo6wi6lzoow)

**Reproduction Line:**
```
./gradlew ":modules:reindex-management:internalClusterTest" --tests "org.elasticsearch.reindex.management.CancelTasksRelocationIT.testCancelByOriginalTaskIdAfterRelocation" -Dtests.seed=76609432576F89A0 -Dtests.locale=rof -Dtests.timezone=US/Alaska -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.reindex.management.CancelTasksRelocationIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testCancelByOriginalTaskIdAfterRelocation'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: slice workers should be registered under the relocated parent
Expected: a collection with size <4>
     but: collection size was <3>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testCancelByOriginalTaskIdAfterRelocation
- [main] 1 consecutive failures in step part1
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 2 failures in test testCancelByOriginalTaskIdAfterRelocation (0.5% fail rate in 410 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/reindex-management/src/internalClusterTest/java/org/elasticsearch/reindex/management/CancelTasksRelocationIT.java
+++ b/modules/reindex-management/src/internalClusterTest/java/org/elasticsearch/reindex/management/CancelTasksRelocationIT.java
@@ -380,19 +380,18 @@ public class CancelTasksRelocationIT extends ESIntegTestCase {
         assertThat("relocated task has a distinct numeric id", relocatedTaskId, not(equalTo(originalTaskId)));
 
         if (slices > 1) {
-            BulkByPaginatedSearchTask.Status status = asInstanceOf(
-                BulkByPaginatedSearchTask.Status.class,
-                clusterAdmin().getTask(new GetTaskRequest().setTaskId(originalTaskId)).actionGet().getTask().getTask().status()
-            );
-            int slicesCompletedOnOriginalTask = toIntExact(status.getSliceStatuses().stream().filter(Objects::nonNull).count());
-
-            assertBusy(
-                () -> assertThat(
+            assertBusy(() -> {
+                BulkByPaginatedSearchTask.Status status = asInstanceOf(
+                    BulkByPaginatedSearchTask.Status.class,
+                    clusterAdmin().getTask(new GetTaskRequest().setTaskId(originalTaskId)).actionGet().getTask().getTask().status()
+                );
+                int slicesCompletedOnOriginalTask = toIntExact(status.getSliceStatuses().stream().filter(Objects::nonNull).count());
+                assertThat(
                     "slice workers should be registered under the relocated parent",
                     listChildrenOf(relocatedTaskId),
                     hasSize(slices - slicesCompletedOnOriginalTask)
-                )
-            );
+                );
+            });
         }
 
         return new RelocatedReindex(survivorNodeName, originalTaskId, relocatedTaskId);

--- a/modules/reindex-management/src/internalClusterTest/java/org/elasticsearch/reindex/management/CancelTasksRelocationIT.java
+++ b/modules/reindex-management/src/internalClusterTest/java/org/elasticsearch/reindex/management/CancelTasksRelocationIT.java
@@ -14,6 +14,7 @@ import org.apache.logging.log4j.Logger;
 import org.elasticsearch.ElasticsearchStatusException;
 import org.elasticsearch.action.ActionFuture;
 import org.elasticsearch.action.admin.cluster.node.tasks.cancel.CancelTasksRequest;
+import org.elasticsearch.action.admin.cluster.node.tasks.get.GetTaskRequest;
 import org.elasticsearch.action.admin.cluster.node.tasks.list.ListTasksResponse;
 import org.elasticsearch.action.get.GetResponse;
 import org.elasticsearch.client.Request;
@@ -24,6 +25,7 @@ import org.elasticsearch.cluster.node.DiscoveryNodeRole;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.core.Strings;
 import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.index.reindex.BulkByPaginatedSearchTask;
 import org.elasticsearch.index.reindex.ReindexAction;
 import org.elasticsearch.index.reindex.ResumeReindexAction;
 import org.elasticsearch.node.ShutdownPrepareService;
@@ -47,11 +49,13 @@ import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 
+import static java.lang.Math.toIntExact;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertNoFailures;
 import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.equalTo;
@@ -376,11 +380,17 @@ public class CancelTasksRelocationIT extends ESIntegTestCase {
         assertThat("relocated task has a distinct numeric id", relocatedTaskId, not(equalTo(originalTaskId)));
 
         if (slices > 1) {
+            BulkByPaginatedSearchTask.Status status = asInstanceOf(
+                BulkByPaginatedSearchTask.Status.class,
+                clusterAdmin().getTask(new GetTaskRequest().setTaskId(originalTaskId)).actionGet().getTask().getTask().status()
+            );
+            int slicesCompletedOnOriginalTask = toIntExact(status.getSliceStatuses().stream().filter(Objects::nonNull).count());
+
             assertBusy(
                 () -> assertThat(
                     "slice workers should be registered under the relocated parent",
                     listChildrenOf(relocatedTaskId),
-                    hasSize(slices)
+                    hasSize(slices - slicesCompletedOnOriginalTask)
                 )
             );
         }

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -570,6 +570,3 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
   method: test {feature:SUBQUERIES}
   issue: https://github.com/elastic/elasticsearch/issues/149681
-- class: org.elasticsearch.reindex.management.CancelTasksRelocationIT
-  method: testCancelByOriginalTaskIdAfterRelocation
-  issue: https://github.com/elastic/elasticsearch/issues/149688

